lighthouse-core/report/html/components/gauge.js

```diff
@@ -0,0 +1,398 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+const template = document.createElement('template');
+template.innerHTML = `
+  <style>
+  :host {
+
+    /* Palette using Material Design Colors
+    * https://www.materialui.co/colors */
+   --color-amber-50: #FFF8E1;
```

Comment:
The dependency chain of variables was surprisingly difficult to unravel, so I just temporarily copied _all_ of them, the actual set is still going to be surprisingly and frustratingly large though :(

Reply:
I'm a little confused here .. would every single component require its own set of all the variables it needs? The component can't make use of variable defined outside of itself?

Reply:
basically, yes. if it's an actual reusable component it would need to contain it's own variables even if they could be *overridden* by our report styles

Reply:
Could we have a "variables" component (which you must include in any page that uses LH components), and remove all (shared) variables from each individual component?